File: android/app/src/main/java/com/privatepensieve/app/screens/PrivacyScreen.kt
package com.privatepensieve.app.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.privatepensieve.app.ui.theme.PensieveColors

/**
 * Privacy Screen — product differentiator.
 * Shows all privacy status items and action buttons.
 */
@Composable
fun PrivacyScreen(modifier: Modifier = Modifier) {
    val scrollState = rememberScrollState()

    Column(
        modifier = modifier
            .fillMaxSize()
            .background(PensieveColors.background)
            .verticalScroll(scrollState)
            .padding(horizontal = 16.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        // Header
        Text(
            text = "Privacy, by design",
            fontSize = 28.sp,
            fontWeight = FontWeight.Bold,
            color = PensieveColors.textPrimary,
            modifier = Modifier.padding(top = 24.dp)
        )

        Text(
            text = "Your memories do not leave this device.",
            fontSize = 14.sp,
            color = PensieveColors.textSecondary,
            modifier = Modifier.padding(top = 4.dp)
        )

        Spacer(modifier = Modifier.height(24.dp))

        // Status list
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(20.dp))
                .background(PensieveColors.surfacePrimary)
        ) {
            PrivacyStatusRow("Account", "Not required")
            HorizontalDivider(color = PensieveColors.border)
            PrivacyStatusRow("Server", "Not used")
            HorizontalDivider(color = PensieveColors.border)
            PrivacyStatusRow("Cloud", "Not used")
            HorizontalDivider(color = PensieveColors.border)
            PrivacyStatusRow("Internet", "Not required")
            HorizontalDivider(color = PensieveColors.border)
            PrivacyStatusRow("AI", "On-device")
            HorizontalDivider(color = PensieveColors.border)
            PrivacyStatusRow("Memories", "Encrypted locally")
            HorizontalDivider(color = PensieveColors.border)
            PrivacyStatusRow("Backup", "Manual encrypted export only")
            HorizontalDivider(color = PensieveColors.border)
            PrivacyStatusRow("Tracking", "None")
        }

        Spacer(modifier = Modifier.height(24.dp))

        // Action buttons
        val actions = listOf(
            "Export encrypted backup",
            "Import backup",
            "Manage offline brain pack",
            "Delete audio recordings",
            "Delete all memories",
            "View privacy promise",
            "View open-source code"
        )

        actions.forEach { action ->
            val isDestructive = action.startsWith("Delete")
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(vertical = 4.dp)
                    .clip(RoundedCornerShape(16.dp))
                    .background(PensieveColors.surfacePrimary)
                    .padding(16.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = action,
                    fontSize = 16.sp,
                    color = if (isDestructive) PensieveColors.accentRed else PensieveColors.textPrimary,
                    modifier = Modifier.weight(1f)
                )
            }
        }

        Spacer(modifier = Modifier.height(32.dp))
    }
}

@Composable
private fun PrivacyStatusRow(label: String, value: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 14.dp)
            .semantics { contentDescription = "$label: $value" },
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(
            text = label,
            fontSize = 16.sp,
            color = PensieveColors.textPrimary,
            modifier = Modifier.weight(1f)
        )
        Text(
            text = value,
            fontSize = 14.sp,
            color = PensieveColors.textSecondary
        )
    }
}

File: ios/PrivatePensieve/Screens/Privacy/PrivacyScreen.swift
// PrivacyScreen.swift
// Private Pensieve AI — iOS
// Privacy dashboard: product differentiator

import SwiftUI

struct PrivacyScreen: View {
    var body: some View {
        NavigationStack {
            ZStack {
                Color.pensieveBackground
                    .ignoresSafeArea()

                ScrollView {
                    VStack(spacing: 24) {
                        // Header
                        VStack(spacing: 8) {
                            Text("Privacy, by design")
                                .font(.largeTitle)
                                .fontWeight(.bold)
                                .foregroundColor(.pensieveTextPrimary)

                            Text("Your memories do not leave this device.")
                                .font(.body)
                                .foregroundColor(.pensieveTextSecondary)
                        }
                        .padding(.top, 24)

                        // Status list
                        VStack(spacing: 0) {
                            PrivacyStatusRow(label: "Account", value: "Not required", icon: "person.slash")
                            Divider().background(Color.pensieveBorder)
                            PrivacyStatusRow(label: "Server", value: "Not used", icon: "server.rack")
                            Divider().background(Color.pensieveBorder)
                            PrivacyStatusRow(label: "Cloud", value: "Not used", icon: "icloud.slash")
                            Divider().background(Color.pensieveBorder)
                            PrivacyStatusRow(label: "Internet", value: "Not required", icon: "wifi.slash")
                            Divider().background(Color.pensieveBorder)
                            PrivacyStatusRow(label: "AI", value: "On-device", icon: "brain")
                            Divider().background(Color.pensieveBorder)
                            PrivacyStatusRow(label: "Memories", value: "Encrypted locally", icon: "lock.fill")
                            Divider().background(Color.pensieveBorder)
                            PrivacyStatusRow(label: "Backup", value: "Manual encrypted export only", icon: "square.and.arrow.up")
                            Divider().background(Color.pensieveBorder)
                            PrivacyStatusRow(label: "Tracking", value: "None", icon: "hand.raised.slash")
                        }
                        .background(Color.pensieveSurfacePrimary)
                        .clipShape(RoundedRectangle(cornerRadius: 20))
                        .padding(.horizontal, 16)

                        // Actions
                        VStack(spacing: 12) {
                            PrivacyActionButton(title: "Export encrypted backup", icon: "square.and.arrow.up") {
                                // TODO: BACKUP-001
                            }
                            PrivacyActionButton(title: "Import backup", icon: "square.and.arrow.down") {
                                // TODO: BACKUP-001
                            }
                            PrivacyActionButton(title: "Manage offline brain pack", icon: "brain") {
                                // TODO: Brain pack manager
                            }
                            PrivacyActionButton(title: "Delete audio recordings", icon: "waveform", isDestructive: true) {
                                // TODO: PRIV-001
                            }
                            PrivacyActionButton(title: "Delete all memories", icon: "trash", isDestructive: true) {
                                // TODO: PRIV-001
                            }
                            PrivacyActionButton(title: "View privacy promise", icon: "doc.text") {
                                // TODO: Show privacy promise
                            }
                            PrivacyActionButton(title: "View open-source code", icon: "chevron.left.forwardslash.chevron.right") {
                                // TODO: Open repository URL
                            }
                        }
                        .padding(.horizontal, 16)
                        .padding(.bottom, 32)
                    }
                }
            }
        }
    }
}

// MARK: - Supporting Views

struct PrivacyStatusRow: View {
    let label: String
    let value: String
    let icon: String

    var body: some View {
        HStack {
            Image(systemName: icon)
                .foregroundColor(.pensieveAccentTeal)
                .frame(width: 28)

            Text(label)
                .font(.body)
                .foregroundColor(.pensieveTextPrimary)

            Spacer()

            Text(value)
                .font(.subheadline)
                .foregroundColor(.pensieveTextSecondary)
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 14)
        .accessibilityElement(children: .combine)
        .accessibilityLabel("\(label): \(value)")
    }
}

struct PrivacyActionButton: View {
    let title: String
    let icon: String
    var isDestructive: Bool = false
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack {
                Image(systemName: icon)
                    .foregroundColor(isDestructive ? .pensieveAccentRed : .pensieveAccentLavender)
                    .frame(width: 28)

                Text(title)
                    .font(.body)
                    .foregroundColor(isDestructive ? .pensieveAccentRed : .pensieveTextPrimary)

                Spacer()

                Image(systemName: "chevron.right")
                    .font(.caption)
                    .foregroundColor(.pensieveTextMuted)
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 14)
            .background(Color.pensieveSurfacePrimary)
            .clipShape(RoundedRectangle(cornerRadius: 16))
        }
        .accessibilityLabel(title)
    }
}

#Preview {
    PrivacyScreen()
        .preferredColorScheme(.dark)
}
